scripts/approximations/main.py

```diff
@@ -0,0 +1,90 @@
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+
+import polynomial
+import rational
+import approximations
+
+# This script does the following:
+# - Computes polynomial and rational approximations of a given function (e^x by default).
+# - Computes (x,y) coordinates for every approximation given the same x coordinates.
+# - Plots the results for rough comparison.
+# The following are the resources used to write the script:
+# https://xn--2-umb.com/22/approximation/
+# https://sites.tufts.edu/atasissa/files/2019/09/remez.pdf
+def main():
+
+    ##############################
+    # 1. Configuration Parameters
+
+    # start of the interval to calculate the approximation on
+    x_start = 0
+    # end of the interval to calculate the approximation on
+    x_end = 3
+    # number of terms in the polynomial approximation / numerator of the rational approximation.
+    num_terms_approximation = 3
+
+    # number of (x,y) coordinates used to plot the resulting approximation.
+    num_points_plot = 10
+    # number of (x,y) coordinates used to plot the highly accurate
+    # equispaced approximation that we treat as "the closest to truth".
+    num_points_plot_accurate = 50000
```

Comment:
Why wouldn't we just plot the actual function were approximating directly?

Reply:
I think the comment is wrong and requires changes.

This is indeed the actual function.

What we do is spread the X-axis into equal intervals. For each, X we compute the true value of Y.

I incorrectly called this "equispaced" because of the similarity to splitting X-axis into intervals with the equispaced approximation.

Reply:
Resolving based on ACK. The comment was corrected